app/lib/task/backend.dart

```diff
@@ -1216,20 +1243,12 @@ List<Version> _versionsToTrack(
   }.nonNulls.where(visibleVersions.contains).toList();
 }
 
-List<String> _updatedDependencies(
-  List<String>? dependencies,
-  List<String>? discoveredDependencies, {
+List<String> _newDependencies(
```

Comment:
Let's document what this method does.

Aside: This may be in another step, but with SQL, we should probably check if we can get away with simpler behavior, e.g. not reading and processing everything, or, maybe updating the dependent package's timestamp through a query.